### DISCUSSION TITLE
Welcome to ray4 Discussions!

### DISCUSSION THREAD

**Discussion by hollasch** [hollasch/ray4#7] [General]
Feel free to ask questions here, comment on related work, or share your 4D visualization project.

**Comment by neverhood311** [comment]
A few years ago I did some undergraduate research on Global Illumination in 4D as well as a Master's Thesis on [General-Purpose 4D Animation](https://scholarsarchive.byu.edu/etd/6968/). I also published a [paper based on my thesis](https://www.researchgate.net/publication/325426062_Fourveo_Integration_of_4D_Animation_into_Conventional_3D_Animation_Workflows). I also created a [handful](https://www.youtube.com/playlist?list=PLP1FVyLCCHYggAyB7QDor84Uj-rJL-6sF) of [animations](https://www.youtube.com/channel/UCHnTdtUrB1L_1Xyid8tyZPg) with my prototypes ([Rayito4D](https://github.com/neverhood311/Rayito-4D/tree/master/Rayito_Stage5_QT_4D) and [Fourveo](http://fourveo.herokuapp.com/)). I even tried to start a [blog devoted to 4D animation](https://fourveo.blogspot.com/).

Since starting my career, I've moved away from such exotic domains within Computer Graphics. However, lately I've been getting the itch to continue the 4D graphics/animation research as a side project. As I find the time, I plan to take my JavaScript prototype of Fourveo and rewrite it as a more robust C++ 4D animation application. I definitely want to see 4D animation become more mainstream and would love to see the standardization of 4D frameworks, file formats, visualization techniques, etc.

I'm very interested to watch the progress of this repo.
